src/main/java/com/book/store/service/impl/OrderServiceImpl.java

```diff
@@ -0,0 +1,122 @@
+package com.book.store.service.impl;
+
+import com.book.store.dto.order.CreateOrderRequestDto;
+import com.book.store.dto.order.OrderDto;
+import com.book.store.dto.order.UpdateOrderStatusDto;
+import com.book.store.dto.orderitem.OrderItemDto;
+import com.book.store.exception.EntityNotFoundException;
+import com.book.store.mapper.OrderItemMapper;
+import com.book.store.mapper.OrderMapper;
+import com.book.store.model.CartItem;
+import com.book.store.model.Order;
+import com.book.store.model.OrderItem;
+import com.book.store.model.ShoppingCart;
+import com.book.store.model.User;
+import com.book.store.repository.order.OrderRepository;
+import com.book.store.repository.shoppingcart.ShoppingCartRepository;
+import com.book.store.repository.user.UserRepository;
+import com.book.store.service.OrderService;
+import java.math.BigDecimal;
+import java.time.LocalDateTime;
+import java.util.List;
+import java.util.Set;
+import java.util.stream.Collectors;
+import lombok.RequiredArgsConstructor;
+import org.springframework.data.domain.Page;
+import org.springframework.data.domain.Pageable;
+import org.springframework.security.core.Authentication;
+import org.springframework.security.core.context.SecurityContextHolder;
+import org.springframework.stereotype.Service;
+import org.springframework.transaction.annotation.Transactional;
+
+@Service
+@RequiredArgsConstructor
+@Transactional
+public class OrderServiceImpl implements OrderService {
+    private final UserRepository userRepository;
+    private final ShoppingCartRepository shoppingCartRepository;
+    private final OrderMapper orderMapper;
+    private final OrderItemMapper orderItemMapper;
+    private final OrderRepository orderRepository;
+
+    @Override
+    public Page<OrderDto> findOrders(Pageable pageable) {
+        ShoppingCart shoppingCart = getShoppingCartByUser();
+        return orderRepository.findAllByUser_Id(shoppingCart.getUser().getId(), pageable)
+                .map(orderMapper::toDto);
+    }
+
+    @Override
+    public OrderDto saveOrder(CreateOrderRequestDto dto) {
+        ShoppingCart shoppingCart = getShoppingCartByUser();
+        if (shoppingCart.getCartItems().isEmpty()) {
+            throw new IllegalStateException("Shopping Cart is empty");
```

Comment:
add parameter to exception
```suggestion
            throw new OrderProcessingException("Shopping Cart is empty");
```